apollos-church-api/src/config.js

```diff
@@ -1,7 +1,19 @@
 import path from 'path';
-import dotenv from 'dotenv/config'; // eslint-disable-line
+import fetch from 'node-fetch';
+import dotenv from "dotenv/config"; // eslint-disable-line
 import ApollosConfig from '@apollosproject/config';
 
 ApollosConfig.loadYaml({
   configPath: path.join(__dirname, '..', 'config.yml'),
 });
+
+// autodetect Apollos plugin
+(async () => {
+  if (!ApollosConfig.ROCK) return;
+  const res = await fetch(
+    `${ApollosConfig.ROCK.URL}/api/Apollos/GetContentChannelItemsByIds?ids=0`
+  );
+  const usePlugin = res.status === 200;
+  if (usePlugin) console.log('Apollos Rock plugin detected!');
+  ApollosConfig.loadJs({ ROCK: { USE_PLUGIN: usePlugin } });
```

Comment:
absolute genius. I wonder if there's an easy way to do this for the rock version as well. i think I checked for an endpoint a while back and came up empty.

Reply:
Also a little nitpicky, but this might be a better endpoint to check

curl 'https://rock.newspring.cc/api/RestControllers/RestControllerNames?includeObsolete=false'

result:

```
["AchievementAttempts","AchievementTypePrerequisites","AchievementTypes","AnalyticsDimCampus","AnalyticsDimFamilyCurrents","AnalyticsDimFamilyHeadOfHouseholds","AnalyticsDimFamilyHistoricals","AnalyticsDimFinancialAccounts","AnalyticsDimFinancialBatches","AnalyticsDimPersonCurrents","AnalyticsDimPersonHistoricals","AnalyticsFactAttendances","AnalyticsFactFinancialTransactions","AnalyticsSourceAttendances","AnalyticsSourceCampus","AnalyticsSourceFamilyHistoricals","AnalyticsSourceFinancialTransactions","AnalyticsSourcePersonHistoricals","Apollos","Assessments","AssessmentTypes","AssetStorageProviders","AttendanceCheckInSessions","AttendanceCodes","AttendanceOccurrences","Attendances","AttributeMatrices","AttributeMatrixItems","AttributeMatrixTemplates","AttributeQualifiers","Attributes","AttributeValueHistoricals","AttributeValues","AuditDetails","Audits","Auth","Auths","BackgroundChecks","Badges","BenevolenceRequestDocuments","BenevolenceRequests","BenevolenceResults","BinaryFileDatas","BinaryFiles","BinaryFileTypes","Blocks","BlockTypes","Books","BookVersions","Campuses","Categories","Chapters","Checkin","CommunicationAttachments","CommunicationRecipients","CommunicationResponses","Communications","CommunicationTemplateAttachments","CommunicationTemplates","ConnectionActivityTypes","ConnectionOpportunities","ConnectionOpportunityCampus","ConnectionOpportunityConnectorGroups","ConnectionOpportunityGroupConfigs","ConnectionOpportunityGroups","ConnectionRequestActivities","ConnectionRequests","ConnectionRequestWorkflows","ConnectionStatus","ConnectionTypes","ConnectionWorkflows","ContentChannelItemAssociations","ContentChannelItems","ContentChannelItemSlugs","ContentChannels","ContentChannelTypes","DataViewFilters","DataViews","DefinedTypes","DefinedValues","Devices","Documents","DocumentTypes","EntityCampusFilters","EntitySetItems","EntitySets","EntityTypes","EventCalendarContentChannels","EventCalendarItems","EventCalendars","EventItemAudiences","EventItemOccurrenceChannelItems","EventItemOccurrenceGroupMaps","EventItemOccurrences","EventItems","ExceptionLogs","FieldTypes","FileBrowser","FinancialAccounts","FinancialBatches","FinancialGateways","FinancialPaymentDetails","FinancialPersonBankAccounts","FinancialPersonSavedAccounts","FinancialPledges","FinancialScheduledTransactionDetails","FinancialScheduledTransactions","FinancialTransactionAlerts","FinancialTransactionAlertTypes","FinancialTransactionDetails","FinancialTransactionImages","FinancialTransactionRefunds","FinancialTransactions","FollowingEventNotifications","FollowingEventSubscriptions","FollowingEventTypes","Followings","FollowingSuggesteds","FollowingSuggestionTypes","GroupDemographicTypes","GroupDemographicValues","GroupFinder","GroupHistoricals","GroupLocationHistoricals","GroupLocationHistoricalSchedules","GroupLocations","GroupMemberAssignments","GroupMemberHistoricals","GroupMemberRequirements","GroupMembers","GroupMemberScheduleTemplates","GroupMemberWorkflowTriggers","GroupRequirements","GroupRequirementTypes","Groups","GroupScheduleExclusions","GroupSyncs","GroupTypeRoles","GroupTypes","Histories","HtmlContents","InteractionChannels","InteractionComponents","InteractionDeviceTypes","Interactions","InteractionSessions","Lava","LavaShortcodes","Layouts","Locations","MergeFields","MergeTemplates","MetaFirstNameGenderLookups","MetaLastNameLookups","MetaNickNameLookups","MetricCategories","MetricPartitions","Metrics","MetricValuePartitions","MetricValues","Mobile","NcoaHistories","NoteAttachments","Notes","NoteTypes","NoteWatches","NotificationRecipients","Notifications","PageContexts","PageRoutes","Pages","PageShortLinks","Pbx","People","PersistedDatasets","PersonalDevices","PersonAlias","PersonDuplicates","PersonPreviousNames","PersonScheduleExclusions","PersonSearchKeys","PersonSignals","PersonTokens","PersonVieweds","PhoneNumbers","PluginMigrations","PrayerRequests","Presence","RegistrationInstances","RegistrationRegistrantFees","RegistrationRegistrants","Registrations","RegistrationTemplateDiscounts","RegistrationTemplateFeeItems","RegistrationTemplateFees","RegistrationTemplateFormFields","RegistrationTemplateForms","RegistrationTemplatePlacements","RegistrationTemplates","RelatedEntities","ReportFields","Reports","RestActions","RestControllers","RobotFile","RocketChat","SC_Misc_Blocks","ScheduleCategoryExclusions","Schedules","Search","ServiceJobHistories","ServiceJobs","ServiceLogs","SignalTypes","SignatureDocuments","SignatureDocumentTemplates","SiteDomains","Sites","SmsActions","SmsPipelines","SmsPipelines","StatementGeneratorFinancialTransactions","StepPrograms","Steps","StepStatus","StepTypePrerequisites","StepTypes","StepWorkflows","StepWorkflowTriggers","Streaks","StreakTypeExclusions","StreakTypes","SystemCommunications","TaggedItems","Tags","UserLogins","Utility","WebFarmNodeLogs","WebFarmNodes","WistiaAccounts","WistiaMedias","WistiaProjects","WorkflowActionFormAttributes","WorkflowActionForms","WorkflowActions","WorkflowActionTypes","WorkflowActivities","WorkflowActivityTypes","Workflows","WorkflowTriggers","WorkflowTypes"]
```

Reply:
yea i like that better

Reply:
this is the response I'm getting with that call

<img width="576" alt="Screen Shot 2021-05-04 at 10 57 35 AM" src="https://user-images.githubusercontent.com/2659478/117023970-94cf1200-acc7-11eb-966f-2e9b7b2a7e79.png">

Reply:
ah, I need the token

Reply:
so weird, including the token still gives a 401